### PR TITLE
18 synthetic returns

### PR DESCRIPTION
**Pull request by kasperjo**
Made some comments about the return prediction function

### CODE REVIEW

**Comment by phschiele** [comment]
@kasperjo It won't make as much of a difference, but I suggest we do not use the synthetic returns for the risk mode. There we can just use a ewma based on realised returns. Making that change now if it's fine with you.

**Comment by kasperjo** [comment]
> @kasperjo It won't make as much of a difference, but I suggest we do not use the synthetic returns for the risk mode. There we can just use a ewma based on realised returns. Making that change now if it's fine with you.

Yes definitely agree! That makes a lot of sense

File: experiments/backtest.py [hunk 1]
```diff
@@ -46,9 +46,8 @@ def run_backtest(
 ) -> tuple[pd.Series, pd.DataFrame]:
     """
     Run a simplified backtest for a given strategy.
-    At time t we use data from t-500 to t-1 to forecast the data and
-    compute the optimal portfolio weights and cash holdings.
-    We then trade to these weights at time t.
+    At time t we use data from t-lookback to t to compute the optimal portfolio
+    weights and then execute the trades at time t.
```
Comment (phschiele) [inline experiments/backtest.py:50]: If we literally include the returns of `t+1` in our forecast, I think the additional complexity of observing prices at `t-1`, then creating the optimal weights and trading at `t` did not make sense. The below changes are a result of this.

File: experiments/backtest.py [hunk 2]
```diff
@@ -62,25 +61,28 @@ def run_backtest(
     post_trade_cash = []
     post_trade_quantities = []
 
-    returns = synthetic_returns(prices).dropna()
-    means = returns.ewm(halflife=125).mean()
-    covariance_df = returns.ewm(halflife=125).cov()
+    returns = prices.pct_change().dropna()
+    means = (
+        synthetic_returns(prices).shift(-1).dropna()
```
Comment (phschiele) [inline experiments/backtest.py:66]: We need `shift(-1)` here, as when queried at `t`, we want to include the (noisy) returns from `t` to `t+1` (this is also why I needed to shorten the backtest by one day.

**Review by phschiele — APPROVED** [review]
@kasperjo Looks good, feel free to merge if you agree with my changes.

I checked throughout the dataset how often the sign of the synthetic return matches the realized one. In the Kahn paper, this was cited as 54%, we have only 51.3%

image


The reason appears to be that our average variance is 0.0004, compared to the 0.0005 in the paper. I think it is not terribly important which exact value we choose here, just something to keep in mind. We could always tweak it to match $\alpha=0.15$ if we wanted.

**Pull request merged**